ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -741,6 +741,13 @@ let migrate_send'  ~__context ~vm ~dest ~live ~vdi_map ~vif_map ~options =
          else acc)
       [] vm_and_snapshots in
 
+  (* Double check that all of the suspend VDIs are all visible on the source *)
+  let localhost = Helpers.get_localhost ~__context in
+  List.iter (fun vdi_mirror ->
+      let sr = Db.VDI.get_SR ~__context ~self:vdi_mirror.vdi in
+      if not (Helpers.host_has_pbd_for_sr ~__context ~host:localhost ~sr)
+      then raise (Api_errors.Server_error (Api_errors.suspend_image_not_accessible, [ Ref.string_of vdi_mirror.vdi ]))) suspends_vdis;
```

Comment:
The advantage of putting it here is that it's after the host selection has been done. There's no logic anywhere to try and make sure that the correct host is selected (nor can we guarantee such as host exists), and it's such a low probability of hitting this problem that I don't want to spend the time adding in code to do the selection correctly. Otherwise we run the risk of incorrectly asserting the operation will fail when it might actually succeed.